### DISCUSSION TITLE
Best way to migrate entire app from SQLite to PostgreSQL

### DISCUSSION THREAD

**Discussion by maxjnq** [epicweb-dev/epic-stack#712] [Q&A]
Hey guys 👋

We are looking to switch our epic-stack app from SQLite to PostgreSQL, and would like to do it the best way possible as we have active users. Any helpful links, or advices on how to do this best? We are using fly.io for hosting. And of course, we would make our version of the Epic Stack with PostgreSQL available to everyone.

Thanks a lot.

Maxime

**Comment by kentcdodds** [comment]
First, I'm curious what's driving the move. Where did SQLite fall short for you?

I would start by making a version of the Epic Stack with postgres so you can get an idea of what the end state looks like and what's required. Then you'll need to determine the ratio of service reliability to migration difficulty you're interested in.

If you're ok with a little data loss, then it'll be much easier because you can ssh into your box, run a script to move everything from the SQLite db to the Postgres db, then deploy a version of the app that uses the postgres db instead. You would lose data added between running the script and deploying the app. You could probably run a secondary script to make up for that difference, but it wouldn't be perfect.

If it's critical for you to have a seamless migration, then you'll want to deploy a version of the app that performs writes to both databases for a while, then run the migration script, then deploy a version that reads from the postgres db only, at which point you can remove the SQLite database. It's a bit more work, but it eliminates issues with data loss.

There may be tools that make doing these things easier, but I've only gone the opposite direction and I was ok with a bit of data loss so this is as much help as I can offer. Good luck!

**Comment by maxjnq** [comment]
Thank you for the quick answer, it's only because of these issues (but I hate to switch).

[10403](https://github.com/prisma/prisma/issues/10403) and all [that](https://github.com/prisma/prisma/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+Timed+out+during+query+execution%22). We ran so many times in those panic errors never able to really understand why, it just doesn't seem solid enough or at least maintained well enough from the Prisma team to fully rely on it. Now with Prisma Pulse not compatible on top, it's too many reasons.

Data loss is completely fine, will go with the first option then and hope that will fix our issues. Now I guess we cannot store images in the db the same way it's done in the current Epic Stack right?

Thank you for your help!

Maxime

**Reply by kentcdodds** [reply]
You can definitely still store images in Postgres, but it'll probably not be very efficient.

I've been planning on investigating Tigris anyway: https://fly.io/docs/reference/tigris/

That's what I'd recommend. I welcome a contribution back to the stack if you decide to go that route!

**Reply by maxjnq** [reply]
Great. And of course, will do it first with the original epic-stack, share it and then implement it for ourselves.

Thanks!

**Comment by warronbebster** [comment]
@maxjnq did you all end up making it work with postgres?

**Reply by Erickpe8** [reply]
@maxjnq, yo también tengo la misma duda y hasta me gustaría saber cómo lo lograron.

**Comment by jsrhodes15** [comment]
Dunno if anyone still needs this, but this is how I did it: https://github.com/jsrhodes15/epic-stack-postgres/tree/main

**Reply by kentcdodds** [reply]
Thanks for sharing!
